platform-api/analytics-api/app/controllers/DeviceController.scala

```diff
@@ -1,36 +1,49 @@
 package controllers
 
-import akka.actor.{ActorRef, ActorSystem, Props}
+import akka.actor.{ActorRef, ActorSystem}
+import appconf.AppConf
 import com.google.inject.Inject
-import javax.inject.Named
-import org.ekstep.analytics.api.service.{DeviceRegisterService, RegisterDevice}
-import org.ekstep.analytics.api.util.{CommonUtil, JSONUtils}
-import play.api.libs.json.Json
+import org.ekstep.analytics.api.service.{RegisterDevice}
+import org.ekstep.analytics.api.util.{ CommonUtil, JSONUtils}
+import play.api.libs.json.{JsValue, Json}
 import play.api.mvc.Action
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class DeviceController @Inject()(system: ActorSystem, @Named("save-metrics-actor") metricsActor: ActorRef) extends BaseController {
+object deviceControllerResponse {
+  val success = JSONUtils.serialize(CommonUtil.OK("analytics.device-register",
+    Map("message" -> s"Device registered successfully")))
+}
+
+class DeviceController @Inject()(system: ActorSystem) extends BaseController {
 
   implicit val ec: ExecutionContext = system.dispatchers.lookup("device-register-controller")
-  private val deviceRegisterServiceAPIActor = system.actorOf(Props(new DeviceRegisterService(metricsActor)))
 
-  def registerDevice(deviceId: String) = Action.async { implicit request =>
-    val body: String = Json.stringify(request.body.asJson.get)
+  def registerDevice(deviceId: String) = Action.async(parse.json) { implicit request =>
+    val deviceRegisterServiceAPIActor = AppConf.getActorRef("deviceRegisterService")
+    val successResponse = deviceControllerResponse.success
+    val body: JsValue = request.body
     // The X-Forwarded-For header from Azure is in the format '61.12.65.222:33740, 61.12.65.222'
     val ipAddr = request.headers.get("X-Forwarded-For").map {
       x =>
         val ipArray = x.split(",")
         if (ipArray.length == 2) ipArray(1).trim else ipArray(0).trim
     }
-    val ip = ipAddr.getOrElse("")
+
+    val headerIP = ipAddr.getOrElse("")
     val uaspec = request.headers.get("User-Agent")
+    val ip_addr = (body \ "request" \ "ip_addr").asOpt[String]
+    val fcmToken = (body \ "request" \ "fcmToken").asOpt[String]
+    val producer = (body \ "request" \ "producer").asOpt[String]
+    val dspec: Option[String] = (body \ "request" \ "dspec").toOption  match {
+      case Some(value) => Option(Json.stringify(value))
```

Comment:
If "dspec" is String, then we don't need to stringify again. Do not use Json.stringify, please use JSONUtils.serialize() to convert an object to a string.